### PR TITLE
boards: arm: mimxrt1170_evkb: add RT1170 EVKB

### PR DESCRIPTION
**Pull request by danieldegrasse**
Add RT1170 EVKB definition to Zephyr. This board is similar to the
RT1170 EVK, but uses a different quadspi flash chip, and includes
an audio connector as well as an additional display connection.

The following features have been tested with this port:
The following features have been verified to function on this board:
- GPIO 
- COUNTER
- CAN
- SPI
- I2C
- PWM
- ADC
- UART 
- DMA
- GPT
- WATCHDOG
- USB
- HWINFO
- DISPLAY
- CAAM
- SDHC
- Multicore boot

The following features are not currently tested/enabled:
- ENET (This board uses an unsupported PHY)
- SAI (Should work, but requires board modification to verify)
- ACMP (Untested)
- FLEXSPI (This board's flash chip is not supported)

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

### CODE REVIEW

File: samples/subsys/ipc/openamp/boards/mimxrt1170_evkb_cm7.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CONFIG_INCLUDE_REMOTE_DIR=y
```
Comment (DerekSnell) [inline samples/subsys/ipc/openamp/boards/mimxrt1170_evkb_cm7.conf:1]:
```suggestion
#
# Copyright 2023, NXP
#
# SPDX-License-Identifier: Apache-2.0
#

CONFIG_INCLUDE_REMOTE_DIR=y
```

File: samples/subsys/ipc/openamp/remote/boards/mimxrt1170_evkb_cm4.conf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+CONFIG_BUILD_OUTPUT_INFO_HEADER=y
```
Comment (DerekSnell) [inline samples/subsys/ipc/openamp/remote/boards/mimxrt1170_evkb_cm4.conf:1]:
```suggestion
#
# Copyright 2023, NXP
#
# SPDX-License-Identifier: Apache-2.0
#

CONFIG_BUILD_OUTPUT_INFO_HEADER=y
```

**Review by DerekSnell — CHANGES_REQUESTED** [review]
Hi @danieldegrasse ,
Thank you for contributing this board.  Overall, this LGTM.  However, we also need to update the board doc page, and clarify both boards are supported.  Since these are in the same board folder and same doc page, we should be clear which features are supported on both boards.  Perhaps the Supported Features table should have separate columns: one for the new `EVKB`  revision, and another the original `EVK` revision.

Thanks

**Comment by danieldegrasse** [comment]
> 

Hi @danieldegrasse ,
Thank you for contributing this board. Overall, this LGTM. However, we also need to update the board doc page, and clarify both boards are supported. Since these are in the same board folder and same doc page, we should be clear which features are supported on both boards. Perhaps the Supported Features table should have separate columns: one for the new EVKB revision, and another the original EVK revision.

Thanks

Thanks for highlighting this @DerekSnell. I have added a table as you suggested, and tracked the status of features on each board. I have also added a "(M7)" specifier for any features that are only validated on the M7 core.

**Review by DerekSnell — APPROVED** [review]
LGTM, @danieldegrasse .  Thanks for updating the doc page

File: samples/drivers/display/boards/mimxrt1170_evkb_cm7.conf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright 2023, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_HEAP_MEM_POOL_SIZE=4194304
```
Comment (mmahadevan108) [inline samples/drivers/display/boards/mimxrt1170_evkb_cm7.conf:7]: WHy is this changed needed for EVKB and not EVK?

**Review by decsny — APPROVED** [review]

**Pull request merged**